### PR TITLE
Feature ETP-753: Implement execution of action buttons

### PR DESCRIPTION
**Pull request by Sankku**
The execution of actions via action buttons involves several components working together to provide a seamless user experience. 
image

### CODE REVIEW

File: packages/MainUI/components/Table/index.tsx [hunk 1]
```diff
@@ -53,6 +55,7 @@ const DynamicTableContent = memo(function DynamicTableContent({ tab }: DynamicTa
     ({ row }: { row: MRT_Row<Record<string, unknown>> }) => ({
       onClick: () => {
         selectRecord(row.original as never, tab);
+        setSelectedRecord(row.original as never);
```
Comment (LuuchoRocha) [inline packages/MainUI/components/Table/index.tsx:58]: I think we could rely in the metadata context and the selected records there, but I will check

File: packages/MainUI/components/Toolbar/GenericProcessButton.tsx [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+      disabled={disabled}
+      title={button.name}
+      data-testid={`process-button-${button.id}`}
+      style={{
+        opacity: disabled ? 0.5 : 1,
+        cursor: disabled ? 'not-allowed' : 'pointer',
+      }}
+    />
+  );
+};
```
Comment (LuuchoRocha) [inline packages/MainUI/components/Toolbar/GenericProcessButton.tsx:18]:
```suggestion

const disabledStyle = {
  opacity: 0.5,
  cursor: 'not-allowed',
};

const enabledStyle = {
  opacity: 1,
  cursor: 'pointer',
}

const GenericProcessButton: React.FC<ProcessButtonProps> = ({ button, onClick, disabled }) => {
  return (
    <IconButton
      onClick={onClick}
      disabled={disabled}
      title={button.name}
      data-testid={`process-button-${button.id}`}
      style={disabled ? disabledStyle : enabledStyle}
    />
  );
};
```

File: packages/MainUI/components/Toolbar/Toolbar.tsx [hunk 3]
```diff
@@ -0,0 +1,274 @@
...
+  const recordContext = useContext(RecordContext);
+  const { executeProcess } = useProcessExecution();
+  const router = useRouter();
+  const { t } = useTranslation();
+
+  const handleAction = (action: string) => {
+    if (action === BUTTON_IDS.NEW) {
+      router.push(`/window/${windowId}/${tabId}/NewRecord`);
+    }
+  };
```
Comment (LuuchoRocha) [inline packages/MainUI/components/Toolbar/Toolbar.tsx:61]: wrap this with useCallback

File: packages/MainUI/components/Toolbar/Toolbar.tsx [hunk 4]
```diff
@@ -0,0 +1,274 @@
...
+          display: 'flex',
+          gap: '0.25rem',
+          background: theme.palette.baselineColor.transparentNeutral[5],
+          borderRadius: '10rem',
+          padding: '0.25rem',
+        },
+      },
+      isItemSelected: !!recordContext?.selectedRecord,
+    };
+  };
```
Comment (LuuchoRocha) [inline packages/MainUI/components/Toolbar/Toolbar.tsx:271]: is there any way you could simplify, refactor into smaller pieces and re-usable ones?

File: packages/MainUI/hooks/useProcessExecution.ts [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+  recordId: BaseFieldDefinition<string>;
+  params?: Record<string, unknown>;
+}
+
+export function useProcessExecution() {
+  const [loading, setLoading] = useState(false);
+  const [error, setError] = useState<Error | null>(null);
+  const { token } = useContext(UserContext);
+
+  const executeProcess = async ({ button, recordId, params = {} }: ExecuteProcessParams): Promise<ProcessResponse> => {
```
Comment (LuuchoRocha) [inline packages/MainUI/hooks/useProcessExecution.ts:18]: wrap this with useCallback

**Review by LuuchoRocha — CHANGES_REQUESTED** [review]

**Review by NazaTarrili-Futit — APPROVED** [review]

**Review by LuuchoRocha — APPROVED** [review]

**Pull request merged**